source/GameInterface/Services/Settlements/Patches/CurrentSiegeStateSettlementPatch.cs

```diff
@@ -0,0 +1,47 @@
+﻿using HarmonyLib;
+using TaleWorlds.CampaignSystem.Settlements;
+using static TaleWorlds.CampaignSystem.Settlements.Settlement;
+using Common.Util;
+using GameInterface.Policies;
+using GameInterface.Services.Settlements.Messages;
+using Common.Messaging;
+using Common;
+using System.Reflection;
+using GameInterface.Extentions;
+
+namespace GameInterface.Services.Settlements.Patches;
+
+/// <summary>
+/// Used to handle CurrentSiegeStatus
+/// </summary>
+
+[HarmonyPatch(typeof(Settlement))]
+public class CurrentSiegeStateSettlementPatch
+{
+
+    [HarmonyPatch(nameof(Settlement.CurrentSiegeState), MethodType.Setter)]
+    [HarmonyPrefix]
+    private static bool CurrentSiegeStatePrefix(ref Settlement __instance, ref SiegeState value)
+    {
+        if (AllowedThread.IsThisThreadAllowed()) return true;
+        if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+
+        if (ModInformation.IsClient) return false;
```

Comment:
Change this to match up to line 42 of https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/GameInterface/Services/Armies/Patches/ArmyLifetimePatches.cs#L42